### PR TITLE
BaSP-TG-09: Employee (Get all - Create)

### PR DESCRIPTION
**Pull request by Licamus**
Creations of the methods get all and create for employee. I modified the index.js, controllers, valitidations and create a new schema for employee

### CODE REVIEW

File: src/controllers/employees.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import Employees from '../models/Employees';
+
+const getAllEmployees = async (req, res) => {
+  try {
+    const employees = await Employees.find();
+
+    return res.status(200).json({
+      message: 'Employee found',
+      data: employees,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/employees.js:10]: I think this prop is not necessary.

File: src/controllers/employees.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    });
+  } catch (error) {
+    return res.json({
+      message: 'An error ocurred',
```
Comment (ignacioaristo) [inline src/controllers/employees.js:14]: we can handle this message in a better way, for example: `message: 'An error ocurred': ${error},`

File: src/controllers/employees.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const getAllEmployees = async (req, res) => {
+  try {
+    const employees = await Employees.find();
+
+    return res.status(200).json({
+      message: 'Employee found',
+      data: employees,
+      error: false,
+    });
+  } catch (error) {
```
Comment (ignacioaristo) [inline src/controllers/employees.js:12]: for consistency, change all `errors` to `err`

File: src/controllers/employees.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      phone: req.body.phone,
+      email: req.body.email,
+      password: req.body.password,
+    });
+
+    const result = await employee.save();
+    return res.status(201).json({
+      message: 'Employee created successfully',
+      data: result,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/employees.js:34]: same as before.

File: src/validations/employees.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    email: Joi.string().email().required(),
+    password: Joi.string().alphanum().required(),
+  });
+
+  const validation = employeesValidation.validate(req.body);
+
+  if (validation.error) {
+    return res.status(400).json({
+      message: `There was an error: ${validation.error.details[0].message}`,
+      data: undefined,
```
Comment (ignacioaristo) [inline src/validations/employees.js:17]: for me, this is not necessary.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/employees.js (hunk 1 above)
Comment (Licamus) [reply src/controllers/employees.js:10]: You are talking about error: false? After the metting of today, i let the error prop stay anyway thinking about a possible future use for the API consumer.  If is not necessary in this stage, tell me and i will delete the prop.

File: src/controllers/employees.js (hunk 4 above)
Comment (Licamus) [reply src/controllers/employees.js:34]: You are talking about error: false? After the metting of today, i let the error prop stay anyway thinking about a possible future use for the API consumer.  If is not necessary in this stage, tell me and i will delete the prop.

File: src/validations/employees.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    email: Joi.string().email().required(),
+    password: Joi.string().alphanum().required(),
+  });
+
+  const validation = employeesValidation.validate(req.body);
+
+  if (validation.error) {
+    return res.status(400).json({
+      message: `There was an error: ${validation.error.details[0].message}`,
+      error: true,
```
Comment (ignacioaristo) [inline src/validations/employees.js:17]: Same here

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/employees.js (hunk 1 above)
Comment (Licamus) [reply src/controllers/employees.js]: Done!

File: src/controllers/employees.js (hunk 2 above)
Comment (Licamus) [reply src/controllers/employees.js:13]: Done!

File: src/controllers/employees.js (hunk 3 above)
Comment (Licamus) [reply src/controllers/employees.js]: Done!

File: src/controllers/employees.js (hunk 4 above)
Comment (Licamus) [reply src/controllers/employees.js]: Done!

File: src/validations/employees.js (hunk 5 above)
Comment (Licamus) [reply src/validations/employees.js]: Done!

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
Remove the file of employee inside data folder. Now we are using the Data Base.

File: src/controllers/employees.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    const employees = await Employees.find();
+
+    return res.status(200).json({
+      message: 'Employee found',
+      data: employees,
+    });
+  } catch (err) {
+    return res.json({
+      message: 'An error ocurred',
+      error: err,
```
Comment (ignacioaristo) [inline src/controllers/employees.js:14]: Remove this error.

File: src/controllers/employees.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+const getAllEmployees = async (req, res) => {
+  try {
+    const employees = await Employees.find();
+
+    return res.status(200).json({
+      message: 'Employee found',
+      data: employees,
+    });
+  } catch (err) {
+    return res.json({
```
Comment (ignacioaristo) [inline src/controllers/employees.js:12]: Status code is missing.

File: src/controllers/employees.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    const result = await employee.save();
+    return res.status(201).json({
+      message: 'Employee created successfully',
+      data: result,
+    });
+  } catch (err) {
+    return res.status(400).json({
+      message: `An error ocurred: ${err}`,
+      error: err,
```
Comment (ignacioaristo) [inline src/controllers/employees.js:37]: remove this line

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/employees.js (hunk 7 above)
Comment (Licamus) [reply src/controllers/employees.js:14]: Done!

File: src/controllers/employees.js (hunk 9 above)
Comment (Licamus) [reply src/controllers/employees.js:37]: Done!

File: src/controllers/employees.js (hunk 8 above)
Comment (Licamus) [reply src/controllers/employees.js:12]: done!

File: src/controllers/employees.js (hunk 2 above)
Comment (ignacioaristo) [reply src/controllers/employees.js:13]: I don't see the change here.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/employees.js (hunk 2 above)
Comment (Licamus) [reply src/controllers/employees.js:13]: It doesn't save, done now!

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**